src/layers/models/layersManager.ts

```diff
@@ -162,13 +161,12 @@ export class LayersManager {
         });
 
         // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-      } else if (jobType === JobAction.UPDATE) {
+      } else if (jobType === JobAction.UPDATE || jobType === JobAction.SWAP_UPDATE) {
         const record = await this.catalog.findRecord(productId, undefined, productType);
         data.metadata.id = record?.metadata.id as string;
-        data.metadata.displayPath = record?.metadata.displayPath as string;
         const recordId = data.metadata.id;
-        const dispalyPath = data.metadata.displayPath;
-        const layerRelativePath = `${recordId}/${dispalyPath}`;
+        let layerRelativePath = '';
+        let useNewTargetFlagInUpdateTasks = this.useNewTargetFlagInUpdateTasks;
```

Comment:
i prefer you set it as empty and set it inside the 'if' statement

let useNewTargetFlagInUpdateTasks = undefined;
if (swap) { 
useNewTargetFlagInUpdateTasks = ...;
} else {
useNewTargetFlagInUpdateTasks = ...;
}

Reply:
done